x-pack/plugins/elastic_assistant/server/lib/attack_discovery/graphs/default_attack_discovery_graph/edges/generate_or_end/index.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+
+import { getGenerateOrEndDecision } from './helpers/get_generate_or_end_decision';
+import { getHasZeroAlerts } from '../helpers/get_has_zero_alerts';
+import type { GraphState } from '../../types';
+
+export const getGeneratOrEndEdge = (logger?: Logger) => {
+  const edge = (state: GraphState): 'end' | 'generate' => {
+    logger?.debug(() => '---GENERATE OR END---');
+    const { anonymizedAlerts } = state;
+
+    const hasZeroAlerts = getHasZeroAlerts(anonymizedAlerts);
+
+    logger?.debug(
+      () =>
+        `generatOrEndEdge is evaluating the following (derived) state:\n${JSON.stringify(
+          {
+            anonymizedAlerts: anonymizedAlerts.length,
+            hasZeroAlerts,
+          },
+          null,
+          2
+        )}`
+    );
+
+    const decision = getGenerateOrEndDecision(hasZeroAlerts);
+
+    logger?.debug(() => `---GENERATE OR END: ${decision}---`);
```

Comment:
```suggestion
    logger?.debug(`generateOrEndEdge finished evaluating the following (derived) state as ${decision}:\n${JSON.stringify(
          {
            anonymizedAlerts: anonymizedAlerts.length,
            hasZeroAlerts,
          },
          null,
          2
        )}`);
```

Or something similar... especially since nothing async is going on here, I think 3 logs is overkill